src/ngraph/runtime/gpu/gpu_memory_manager.cpp

```diff
@@ -62,9 +63,15 @@ runtime::gpu::GPUMemoryManager::~GPUMemoryManager()
 
 void runtime::gpu::GPUMemoryManager::allocate()
 {
+    if (m_open_allocators)
+    {
```

Comment:
I feel this can not ensure all reservations is done.

Reply:
for one allocator, you can multiple reserve or no reserve.

Reply:
is this check necessary?

Reply:
This is essentially a liveness check to ensure that the pass::MemoryManager has freed all of its allocations prior to actually making real allocations on the device. It doesn't mean there won't be any more reservation, it's just that there cannot be any that are _in progress_. In other words, a new allocator must be made. 

This way, the external function tensor pool workspace is the first allocation reserved, and then all kernel reservations are "on top" of that, and when the external function memory pool is freed, the kernel workspaces better have already been freed. That is essentially what this check is on, that the reservation tree was release correctly.

This is a bit confusing though and so I replaced the check with the following equivalent check
  ```  if (m_workspace_manager.get_node_list().size() != 1)  ```. Thus the open allocators variable isn't needed and so is removed.